libbeat/metric/system/diskio/diskstat_linux.go

```diff
@@ -52,6 +55,28 @@ func (stat *IOStat) OpenSampling() error {
 	return stat.curCPU.Get()
 }
 
+// a few of the diskio counters are actually 32-bit on the kernel side, which means they can roll over fairly easily.
+// Here we try to reconstruct the values by calculating the pre-rollover delta from unt32 max, then adding.
+// If you want to get technical, this could be a tad unsafe, as we don't actually have any way of knowing if the word size changes in a future kernel, and we've rolled over at UINT64_MAX
+
+// See https://docs.kernel.org/admin-guide/iostats.html and https://github.com/torvalds/linux/blob/master/block/genhd.c diskstats_show()
+func returnOrFixRollover(current, prev uint64) uint64 {
+	var maxUint32 uint64 = math.MaxUint32 //4_294_967_295 Max value in uint32/unsigned int
```

Comment:
`math.MaxUint32` is an untyped constant, so this should not be necessary.

Reply:
Yah, did it as a separate variable more in the hopes of making the logic a little easier to follow.

Reply:
I'm not sure that it does; the point of having these as untyped was exactly to allow this kind of use.